### PR TITLE
Switch to AssertJ.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
This is a pretty naive conversion; we could probably do another pass to take better advantage of AssertJ features. In follow-up!

File: wire-compiler/src/test/java/com/squareup/wire/CommandLineOptionsTest.java [hunk 1]
```diff
@@ -10,38 +10,35 @@
...
 
     options = new CommandLineOptions("--java_out=baz/qux");
-    assertEquals(new File("baz/qux"), options.javaOut);
+    assertThat(options.javaOut).isEqualTo(new File("baz/qux"));
   }
 
   @Test public void sourceFileNames() throws Exception {
     CommandLineOptions options = new CommandLineOptions();
-    assertTrue(options.sourceFileNames.isEmpty());
+    assertThat(options.sourceFileNames.isEmpty()).isTrue();
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/CommandLineOptionsTest.java]: Future refactor: replace `assertThat(foo.isEmpty()).isTrue()` with `assertThat(foo).isEmpty()`.

Comment (swankjesse) [inline wire-compiler/src/test/java/com/squareup/wire/CommandLineOptionsTest.java]: Yes, there's a few of these.

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.java [hunk 2]
```diff
@@ -143,14 +141,15 @@ public ProtoFile parse(String filename) throws IOException {
...
     }
   }
 
   @Test public void testNoPackageNameIsLegal() {
     Map<String, String> output = compile("message Simple { optional int32 f = 1; }");
-    assertTrue(output.containsKey(".Simple"));
+    assertThat(output).containsKey(".Simple");
     // Output should not have a 'package' declaration.
-    assertFalse(output.get(".Simple").contains("package"));
+    assertThat(output.get(".Simple").contains("package")).isFalse();
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.java]: Future refactor: replace both with `assertThat(output.get(".Simple")).doesNotContain("package")` (the `null` check comes free).

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java [hunk 3]
```diff
@@ -47,7 +46,7 @@ private File makeTestDirectory(String path) {
     dir.mkdir();
     cleanup(dir);
     List<String> filesBefore = getAllFiles(dir);
-    assertEquals(0, filesBefore.size());
+    assertThat(filesBefore.size()).isEqualTo(0);
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java]: Future refactor: `.hasSize`

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java [hunk 4]
```diff
@@ -84,7 +83,9 @@ private void testProto(String[] sources, String[] outputs,
     invokeCompiler(args.toArray(new String[args.size()]));
 
     List<String> filesAfter = getAllFiles(testDir);
-    assertEquals(filesAfter.toString(), outputs.length, filesAfter.size());
+    assertThat(filesAfter.size())
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java]: `hasSize`

File: wire-runtime/src/test/java/com/squareup/wire/CustomOptionsTest.java [hunk 5]
```diff
@@ -35,59 +35,57 @@
...
-        option_one.qux);
-    Assert.assertEquals(2, option_one.fred.size());
-    Assert.assertEquals(123.0F, option_one.fred.get(0), 0.0000001);
-    Assert.assertEquals(321, option_one.fred.get(1), 0.0000001);
-    Assert.assertEquals(456.0, option_one.daisy, 0.0000001);
+    assertThat(expected).isEqualTo(-1L);
+    assertThat(option_one.qux).isEqualTo(Long.valueOf(expected));
+    assertThat(option_one.fred.size()).isEqualTo(2);
+    assertThat(option_one.fred.get(0)).isEqualTo(123.0F, Offset.offset(0.0000001f));
+    assertThat(option_one.fred.get(1)).isEqualTo(321f, Offset.offset(0.0000001f));
```
Comment (JakeWharton) [inline wire-runtime/src/test/java/com/squareup/wire/CustomOptionsTest.java]: Cool

**Comment by swankjesse** [comment]
Follow ups: https://github.com/square/wire/pull/260

**Pull request merged**